packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -497,7 +499,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
 
   @autobind
   private _onMenuKeyDown(ev: React.KeyboardEvent<HTMLElement>) {
-    if (ev.which === KeyCodes.down) {
+    if (this.props.allowDirectionalMenuTrigger && ev.which === this._getDirectionKeyCode()) {
```

Comment:
>allowDirectionalMenuTrigger [](start = 19, length = 27)

I looked at the PR and was thinking if we actually need this boolean. Just the presence of the menuTriggerDirection is enough to confirm that the consumer wants to controll the  key to use. No?

Reply:
What if we renamed this prop to `menuTriggerKeyCode` and simply use the `KeyCode` enum as its type?

Reply:
@manishgarg1 , My reasoning is that, since the current default behavior is to allow triggering menu with arrow down, I wanted to also enable it and set the default key to be arrow down. Hence, the bool value was needed to explicitly turn it off. I can of course make the prop `KeyCodes | null`, then disable it when it's set to null. But that sounds unintuitive when not specifying the prop defaults to arrow down.

@dzearing , I thought about that, but I personally felt being able to assign a random key to trigger a menu sounded odd, especially when pressing Enter/Space on a button already triggers a menu. But, I don't have problems with using `KeyCodes` though. Is that still preferable?

Reply:
@ekawatani, exactly as you said. I would prefer "KeyCodes | undefined" as the API signature. And remove the bool. I will confirm with David if he is good with that.

Reply:
@manishgarg1 , Thanks. I can remove the bool, and convert the other prop to use `KeyCode` as David suggested earlier. Just making sure, but we still want to maintain the default behavior to be keydown, correct? So, to disable the custom menu trigger would be to manually assign `null` to the prop.

Let me know if this is what we want, and I'll send an update.

Reply:
Hi @ekawatani , Yes we would like the default behaviour to be keydown. The default value for your new property would be 'undefined' if the consumer code does not set that property. Does that explain the expected behaviour?

Further, just rename the prop to `menuTriggerKeyCode` as David mentioned above.